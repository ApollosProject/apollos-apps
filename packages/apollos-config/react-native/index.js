import NativeConfig from 'react-native-config';

class Config {
  config = {
    ...NativeConfig,
  };

  constructor() {
    this.attachConfigToClass(this.config);
  }

  loadJs(data = {}) {
    Object.keys(data).forEach((key) => {
      if (this.config[key] && typeof this.config[key] === 'object') {
        // Deep merge defaults if they exist.
        this.config[key] = Object.assign(this.config[key], data[key]);
      } else {
        // Otherwise attach results to config.
        this.config[key] = data[key];
      }
    });
    this.attachConfigToClass(this.config);
  }

  // Allows you to do Config.PROPERTY
  attachConfigToClass(config) {
    Object.keys(config).forEach((key) => {
      Object.defineProperty(this, key, {
        enumerable: true,
        configurable: true,
        writable: true,
        value: config[key],
      });
    });
  }
}

const config = new Config();

export default config;
