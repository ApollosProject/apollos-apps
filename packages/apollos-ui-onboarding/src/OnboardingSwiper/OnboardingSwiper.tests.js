import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import renderer from 'react-test-renderer';
import { Providers } from '../testUtils';
import OnboardingSwiper from '.';

describe('The onboarding swiper', () => {
  it('should render with a fragment as a child', () => {
    const tree = renderer.create(
      <Providers>
        <OnboardingSwiper>
          {({ swipeForward, scrollBy }) => (
            <>
              <TouchableOpacity onPress={swipeForward}>
                <Text>1</Text>
              </TouchableOpacity>
              <TouchableOpacity onPress={swipeForward}>
                <Text>2</Text>
              </TouchableOpacity>
              <TouchableOpacity key="3" onPress={() => scrollBy(-1)}>
                <Text>3</Text>
              </TouchableOpacity>
            </>
          )}
        </OnboardingSwiper>
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render with an array as a child', () => {
    const tree = renderer.create(
      <Providers>
        <OnboardingSwiper>
          {({ swipeForward, scrollBy }) => [
            <TouchableOpacity key="1" onPress={swipeForward}>
              <Text>1</Text>
            </TouchableOpacity>,
            <TouchableOpacity key="2" onPress={swipeForward}>
              <Text>2</Text>
            </TouchableOpacity>,
            <TouchableOpacity key="3" onPress={() => scrollBy(-1)}>
              <Text>3</Text>
            </TouchableOpacity>,
          ]}
        </OnboardingSwiper>
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
});
