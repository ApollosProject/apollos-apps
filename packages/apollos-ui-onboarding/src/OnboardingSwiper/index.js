import React, { Component } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import * as ReactIs from 'react-is';

import Swiper from 'react-native-swiper';
import { styled } from '@apollosproject/ui-kit';

import { SafeAreaView } from 'react-navigation';

const dotStyles = ({ theme }) => ({
  width: theme.sizing.baseUnit / 2,
  height: theme.sizing.baseUnit / 2,
  borderRadius: theme.sizing.baseUnit / 4,
  margin: theme.sizing.baseUnit / 4,
});

const forceInset = {
  bottom: 'always',
};

const PaginationDot = styled(
  ({ theme }) => ({
    backgroundColor: theme.colors.background.inactive,
    ...dotStyles({ theme }),
  }),
  'ui-onboarding.OnboardingSwiper.PaginationDot'
)(View);

const PaginationDotActive = styled(
  ({ theme }) => ({
    backgroundColor: theme.colors.action.primary,
    ...dotStyles({ theme }),
  }),
  'ui-onboarding.OnboardingSwiper.PaginationDotActive'
)(View);

class OnboardingSwiper extends Component {
  static navigationOptions = () => ({
    title: 'Onboarding',
    header: null,
    gesturesEnabled: false,
  });

  static propTypes = {
    children: PropTypes.func.isRequired,
  };

  swiper = null;

  // Creates ref to Swiper to be passed as a prop to children.
  setSwiperRef = (r) => {
    this.swiper = r;

    return this.swiper;
  };

  // Advance swiper 1 slide. See Swiper documentation for scrollBy details. https://github.com/leecade/react-native-swiper#methods
  swipeForward = () => this.swiper.scrollBy(1);

  scrollBy = (...args) => this.swiper.scrollBy(...args);

  render() {
    const children = this.props.children({
      swipeForward: this.swipeForward,
      scrollBy: this.scrollBy,
    });

    // The user can pass either a set of components in a fragment, or an array of components.
    // If the uses a fragment, we need to remove the fragment and render the components directly.
    // Otherwise the slider will treat the fragment as a single slide.
    let slides = children;
    if (ReactIs.typeOf(children) === ReactIs.Fragment) {
      slides = children.props.children;
    }

    return (
      <Swiper
        loadMinimal
        loop={false}
        /* Disables swipe gestures. We currently we dont display a back button so this is our
         * only back navigation option. */
        // scrollEnabled={false}
        showsButtons={false}
        ref={this.setSwiperRef}
        renderPagination={this.renderPagination}
        activeDot={
          <SafeAreaView forceInset={forceInset}>
            <PaginationDotActive />
          </SafeAreaView>
        }
        dot={
          <SafeAreaView forceInset={forceInset}>
            <PaginationDot forceInset={forceInset} />
          </SafeAreaView>
        }
        {...this.props}
      >
        {slides}
      </Swiper>
    );
  }
}

export default OnboardingSwiper;
