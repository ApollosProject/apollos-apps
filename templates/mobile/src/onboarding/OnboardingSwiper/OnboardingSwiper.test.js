import { TouchableOpacity, Text } from 'react-native';
import renderer from 'react-test-renderer';
import OnboardingSwiper from './index';

describe('The onboarding swiper', () => {
  it('should render with a fragment as a child', () => {
    const tree = renderer.create(
      <OnboardingSwiper>
        {({ swipeForward, scrollBy }) => (
          <>
            <TouchableOpacity onPress={swipeForward}>
              <Text>1</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={swipeForward}>
              <Text>2</Text>
            </TouchableOpacity>
            <TouchableOpacity key="3" onPress={() => scrollBy(-1)}>
              <Text>3</Text>
            </TouchableOpacity>
          </>
        )}
      </OnboardingSwiper>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render with an array as a child', () => {
    const tree = renderer.create(
      <OnboardingSwiper>
        {({ swipeForward, scrollBy }) => [
          <TouchableOpacity key="1" onPress={swipeForward}>
            <Text>1</Text>
          </TouchableOpacity>,
          <TouchableOpacity key="2" onPress={swipeForward}>
            <Text>2</Text>
          </TouchableOpacity>,
          <TouchableOpacity key="3" onPress={() => scrollBy(-1)}>
            <Text>3</Text>
          </TouchableOpacity>,
        ]}
      </OnboardingSwiper>
    );

    expect(tree).toMatchSnapshot();
  });
});
