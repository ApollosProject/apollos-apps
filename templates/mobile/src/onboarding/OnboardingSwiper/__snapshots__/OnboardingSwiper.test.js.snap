// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The onboarding swiper should render with a fragment as a child 1`] = `
<Swiper
  activeDot={
    <SafeAreaView>
      <Unknown />
    </SafeAreaView>
  }
  dot={
    <SafeAreaView>
      <Unknown />
    </SafeAreaView>
  }
  loadMinimal={true}
  loop={false}
  onIndexChanged={[Function]}
  showsButtons={false}
  userVersion={0}
/>
`;

exports[`The onboarding swiper should render with an array as a child 1`] = `
<Swiper
  activeDot={
    <SafeAreaView>
      <Unknown />
    </SafeAreaView>
  }
  dot={
    <SafeAreaView>
      <Unknown />
    </SafeAreaView>
  }
  loadMinimal={true}
  loop={false}
  onIndexChanged={[Function]}
  showsButtons={false}
  userVersion={0}
/>
`;
